test/assert-test.js

```diff
@@ -1649,4 +1649,45 @@ describe("assert", function () {
                           "    actual = { foo: 1 }");
         });
     });
+
+    if (typeof Symbol === "function") {
+        describe("with symbol method names", function () {
+            function setupSymbol(symbol) {
```

Comment:
Could you please simplify the logic by making it more functional and more explicit? It seems needlessly complex right now :smiley:

- change the setup method
- remove the `bind`  call in line 1662
- simplify `this.message` function (see below) and rename it
- beforeEach => before
- improve test descriptions